plugin/al/usdmayautils/AL/usdmaya/utils/DgNodeHelper.cpp

```diff
@@ -5645,17 +5649,24 @@ MStatus DgNodeHelper::copyDynamicAttributes(MObject node, UsdPrim& prim, Animati
       case MFn::kFloatMatrixAttribute:
       case MFn::kMatrixAttribute:
         {
+          UsdAttribute usdAttr = prim.CreateAttribute(attributeName, !isArray ? SdfValueTypeNames->Matrix4d : SdfValueTypeNames->Matrix4dArray);
+          const auto* converter = Converter::find(plug, usdAttr);
+          if(converter) {
+            ConverterArgs args;
+            converter->convert(plug, usdAttr, args);
+            usdAttr.SetCustom(true);
+            break;
+          }
+            
           if(!isArray)
```

Comment:
Hi Krystian - just to give my questions a bit more concrete footing, this is probably the piece that most confused me.  In this situation, we're ONLY dealing with Matrix4d, and Matrix4dArray types... both of which (according to [your doc](https://github.com/Autodesk/maya-usd/blob/kxl_adsk/one_converter_with_examples/lib/mayaUsd/utils/converter.cpp#L34)), should already be supported.  So I wasn't clear on in which situations `if(converter)` would be expected to return False, or why we would need to keep the old code.

Is it just that you THINK it should handle all the matrix conversions in this particular branch/scope... but you weren't SURE, and that's why you leave the check (and leave the old code in place)? And you wanted AL to confirm that this should work before removing the "old" stuff? Or are there in fact situations where you know/expect `if(converter)` to currently return False?

Reply:
Commit 2e9beee is here to start the conversation about the adoption and make sure we all see the places where we currently do type conversion. This will help us validate the interface I'm proposing.
And commit 2e9beee isn't aiming at proposing the final injection points or strategy for adoption...and this is why I mentioned it won't be returned.

In this particular case, you are commenting on, my thinking was that the final solution shouldn't require this huge `switch`. Your question is valid...and I would as well expect we don't have to fall back to old code.